Source/Scene/ImageryLayerCollection.js

```diff
@@ -389,6 +371,136 @@ ImageryLayerCollection.prototype.pickImageryLayerFeatures = function (
   // Pick against all attached imagery tiles containing the pickedLocation.
   var imageryTiles = pickedTile.data.imagery;
 
+  return [pickedTile, imageryTiles, pickedLocation];
+};
+
+/**
+ * Determines the imagery layers that are intersected by a pick ray. To compute a pick ray from a
+ * location on the screen, use {@link Camera.getPickRay}. This function uses a rectangular partition
+ * approach which is not pixel perfect. Thus, it will pick imagery in several situations where it
+ * should not (transparent imagery, imagery cutout, etc.).
+ *
+ * @param {Ray} ray The ray to test for intersection.
+ * @param {Scene} scene The scene.
+ * @return {ImageryLayer[]|undefined} An `ImageryLayerCollection` that includes all of
+ *                                 the layers that are intersected by a given pick ray. Undefined if
+ *                                 no layers are selected.
+ *
+ */
+ImageryLayerCollection.prototype.pickImageryLayers = function (ray, scene) {
+  var locationData = ImageryLayerCollection.prototype.pickImageryLayersHelper(
+    ray,
+    scene
+  );
+  var pickedTile, imageryTiles, pickedLocation;
+  if (!defined(locationData)) {
+    return undefined;
+  }
+
+  pickedTile = locationData[0];
+  imageryTiles = locationData[1];
+  pickedLocation = locationData[2];
+
+  var imageryLayers = [];
+  for (var i = imageryTiles.length - 1; i >= 0; --i) {
+    var terrainImagery = imageryTiles[i];
+    var imagery = terrainImagery.readyImagery;
+    if (!defined(imagery)) {
+      continue;
+    }
+    var provider = imagery.imageryLayer.imageryProvider;
+    if (!defined(provider.pickFeatures)) {
+      continue;
+    }
```

Comment:
This check is is not needed for `pickImageryLayers` since `pickImageryLayers` isn't concerned about the features.